packages/angular/cli/src/commands/mcp/tools/start-devserver.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { z } from 'zod';
+import { DevServer, LocalDevServer } from '../dev-server';
+import { Host, LocalWorkspaceHost } from '../host';
+import { createStructureContentOutput } from '../utils';
+import { McpToolContext, McpToolDeclaration, declareTool } from './tool-registry';
+
+const startDevserverToolInputSchema = z.object({
+  project: z
+    .string()
+    .optional()
+    .describe(
+      'Which project to serve in a monorepo context. If not provided, serves the default project.',
+    ),
+  port: z.number().optional().describe('Port to listen on.'),
+});
+
+export type StartDevserverToolInput = z.infer<typeof startDevserverToolInputSchema>;
+
+const startDevserverToolOutputSchema = z.object({
+  message: z.string().describe('A message indicating the result of the operation.'),
+});
+
+export type StartDevserverToolOutput = z.infer<typeof startDevserverToolOutputSchema>;
+
+export const DEFAULT_PROJECT_KEY = '<default>';
```

Comment:
Bug: I think the default project still has a name in the `angular.json`, and we might want to use that to avoid duplicating devservers. For example if I ask:

```
Start a devserver. (AI uses the default project with the "<default>" key)
Start a devserver for `my-foo`. (AI uses `my-foo` as the project name, which might be the same project as the default)
```

Ideally we would probably look up the actual name of the default project in the `angular.json` and use that as the key. We can potentially fix this in a future PR and is unlikely to be a major issue, but just something to consider.

Reply:
I think everyone's a little different. I prefer issues as I find those more discoverable and actionable than `TODO` comments, but there's no real standard process.